### DISCUSSION TITLE
Trying to embed a JetEngine shortcode but it fails

### DISCUSSION THREAD

**Discussion by BrettHillMVP** [add2cal/add-to-calendar-button#619] [Q&A]
I'm using Jetengine and custom CPT called Events.
It would be HUGE if I could get the add-to-calendar-button arguments dynamically built.
Very close but not there.

In testing, using the shortcode {sc_start}jet_engine_data{sc_end} works well for dynamically adding the Post title to the ICS calendar entry title in a listing of all events.
BUT, if I want to add any other data that requires additional arguments in the shortcode, it fails to render the ICS properly.
For example
description={sc_start}jet_engine_data dynamic_field_source="meta" dynamic_field_post_meta="_event-description"{sc_end}
This is properly formed based on the Jetengine shortcode engine, but improperly renders the event description in the iCAL file as the text _'[jet_engine_data"_ as shown below.

As the test with only "jet_engine_data" works fine, but the one with a space after jet_engine_data does not, I wonder if the space that is needed to specify the specific custom field is causing the button shortcode parser to halt? If so, the only fix is in the source. 

![2024-08-24_14-33-42](https://github.com/user-attachments/assets/5b6e5011-1470-48bb-81f5-2e537d7263a6)

Unless this specific to my install, this is a pretty big problem for implementation on all jetengine sites. 
Thoughts?

**Comment by jekuer** [comment]
What does the shortcode print?
Might be that you are missing some quotation marks here? -> `description="{sc_start}jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'{sc_end}"`

**Comment by BrettHillMVP** [comment]
Hi,

The shortcode rendered through adto-calendar renders  the calendar descirption as "[jet_engine_data" and nothing else.
The shortcode itself is provided by the Shortcode builder in JetEngine, and is direct copy and paste.
The same shortcode when putting [jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'] on the listing itself directly, it renders the text field properly with even event description showing.

I tested this with a different field and got the same result. See below.

Also, I tried creating a shortcode for this shortode using WPCODE. I was hoping they would let me produce a shortcode that has no spaces. But alas, all the assigned codes have a space. The one I got was [wpcode id="8851"]. That adding that to the button elements results in a simialr output "[wpcode" and nothing else.  They do have a feature that allows you to assign your own codes names, but its only in their $200 annualy price. Sigh.

It seems like this ought to work.

I am not a developer, but the fact that in both cases, the output of both of these instances stops at the first instance of the a space in the embedded shortcode, could be meaningful.

 In particular, this seems informative since the one shortcode I can use for this CPT which is simply {sc_start}jet_engine_data{sc_end} works fine.

Not that the tested item reference below was
description={sc_start}jet_engine_data dynamic_field_source="meta" dynamic_field_post_meta="_event-description"{sc_end} which as metnioned is directly from the shortcode generator in jet.

Thanks,

[cid:ca44faec-2f8d-4322-9f0a-f7361573c9e5]'

[cid:a86c235a-75ec-4bab-bd3b-8de547690bca]
[cid:7fb2f435-fb15-4622-8ccb-306158c39401]

________________________________

**Reply by jekuer** [reply]
Hi,

the images are nto included in the post (usually happens when replying via email).
Did you try what I proposed? Add quotation marks.

**Comment by BrettHillMVP** [comment]
Where?

Brett Hill
Mindful-Somatic Coach and Coach Trainer
Connect on Facebook<https://www.facebook.com/coachbretthill?mibextid=ZbWKwL>, LinkedIn<https://www.linkedin.com/company/themindfulcoachmethod/>
________________________________

**Reply by jekuer** [reply]
`description="{sc_start}jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'{sc_end}"`

`description=""` <- you where not using them in your code above.

**Comment by BrettHillMVP** [comment]
Hi,
BTW, thanks very much for y9ur help.

description="{sc_start}jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'{sc_end}"
Results in no content being in the resulting calendar entry.

You can see the The page is here:
https://mindfulcoachcommunity.com/events/
The "description" is mapped to the field _event-description which you can read on the event listing.

This a copy of the full shortcode for the add2cal

[add-to-calendar-button name="{sc_start}jet_engine_data{sc_end}" description="{sc_start}jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'{sc_end}" startDate="2024-08-26" startTime="10:15" endTime="23:30" timeZone="America/Los_Angeles" location="World Wide Web" options="'Apple','Google','iCal','Outlook.com','Microsoft 365','Microsoft Teams','Yahoo'"]

Thanks
Brett

________________________________

**Comment by BrettHillMVP** [comment]
Hi @jekuer ,
In response to your inquiry, after adding the quotations, as you kindly advised, there is no "description" field in the iCAL download. As mentioned, you can verify the functionality at https://mindfulcoachcommunity.com/events/

To verify the proper setup, the first attachment shows the shortcode for the field. 
The next shows the targeted field has content
And finally, the button shortcode is 
[add-to-calendar-button name="{sc_start}jet_engine_data{sc_end}" description="{sc_start}jet_engine_data dynamic_field_source='meta' dynamic_field_post_meta='_event-description'{sc_end}" startDate="2024-08-26" startTime="10:15" endTime="23:30" timeZone="America/Los_Angeles" location="World Wide Web" options="'Apple','Google','iCal','Outlook.com','Microsoft 365','Microsoft Teams','Yahoo'"]

![2024-08-26_18-14-32](https://github.com/user-attachments/assets/56f41522-a713-41fe-a981-76210ebd9e9c)
![2024-08-26_18-13-11](https://github.com/user-attachments/assets/d4301c18-47c9-4ca1-9197-8f1120089488)

Please, if you would be so kind as to advise,  I need to get this working.
Best to you,
Brett

**Comment by BrettHillMVP** [comment]
You might be interested to know this.

I am able to get this to work, but only by creating a shortcode that has no spaces.

The shortcodes for the meta field created by jet-engine or WPCODE, both have spaces in the shortcode between the [].

But by referencing the exact same field and assigning it shortcode such as 'event-description', the plug in handles that as expected.

Thanks


________________________________

**Reply by jekuer** [reply]
Hi Brett,
This is not about the spaces!
What you observed is (99%) just a coincidence. Please check the different output generated by the different shortcodes. Maybe set the description to something simple like "abc" for testing. Put the shortcode next to the button to see what it prints outside of the button.

Btw: Just tested the first example on your website and it is delivering some description.

**Comment by BrettHillMVP** [comment]
Ok.
Thanks.
It's is delivering now because of the shortcodes I added.
I have done all the tests you suggested of course.
And have of course validated the shortcodes work outside of the button with no suprises.

At this point, I'm well on my to work around the problem.
But it's 100% reproduceable on this site. I can create a staging site for you to look at it if you wish.
But in 3 days of testing, I have had no success with any jetengine shortcode for CPT field other than those I created manually that have only one word.

I am curious about how to present a proper Timzone to your plugin. Do you have any recommendations that? Given that user enters some time zone for their event - what are you looking for in the timeZone element?

thanks
brett

________________________________
